clients/admin-ui/src/features/privacy-requests/configuration/MailgunEmailConfiguration.tsx

```diff
@@ -0,0 +1,156 @@
+  useGetMessagingConfigurationDetailsQuery,
+} from "~/features/privacy-requests/privacy-requests.slice";
+
+const MailgunEmailConfiguration = () => {
+  const { successAlert } = useAlert();
+  const [configurationStep, setConfigurationStep] = useState("");
+  const { handleError } = useAPIHelper();
+  const { data: messagingDetails } = useGetMessagingConfigurationDetailsQuery({
+    type: "mailgun",
+  });
+  const [createMessagingConfiguration] =
+    useCreateMessagingConfigurationMutation();
+  const [createMessagingConfigurationSecrets] =
+    useCreateMessagingConfigurationSecretsMutation();
+
+  const handleMailgunConfiguration = async (value: { domain: string }) => {
+    const result = await createMessagingConfiguration({
+      type: "mailgun",
+      details: {
+        is_eu_domain: "false",
+        domain: value.domain,
+      },
+    });
+
+    if (isErrorResult(result)) {
+      handleError(result.error);
+    } else {
+      successAlert(
+        `Mailgun email successfully updated. You can now enter your security key.`
+      );
+      setConfigurationStep("apiKey");
+    }
+  };
+
+  const handleMailgunAPIKeyConfiguration = async (value: {
+    api_key: string;
+  }) => {
+    const result = await createMessagingConfigurationSecrets({
+      mailgun_api_key: value.api_key,
+    });
+
+    if (isErrorResult(result)) {
+      handleError(result.error);
+    } else {
+      successAlert(`Mailgun security key successfully updated.`);
+    }
+  };
+
+  const initialValues = {
+    domain: messagingDetails?.details.domain ?? "",
+  };
+
+  const initialAPIKeyValue = {
+    api_key: messagingDetails?.key ?? "",
+  };
+
+  return (
+    <Box>
+      <Heading fontSize="md" fontWeight="semibold" mt={10}>
+        Mailgun messaging configuration
+      </Heading>
+      <Stack>
+        <Formik
+          initialValues={initialValues}
+          onSubmit={handleMailgunConfiguration}
+        >
+          {({ isSubmitting, resetForm }) => (
+            <Form>
+              <Stack mt={5} spacing={5}>
+                <CustomTextInput
+                  name="domain"
+                  label="Domain"
+                  placeholder="Enter domain"
+                />
+              </Stack>
+              <Box mt={10}>
+                <Button
+                  onClick={() => resetForm()}
+                  mr={2}
+                  size="sm"
+                  variant="outline"
+                >
+                  Cancel
+                </Button>
+                <Button
+                  isDisabled={isSubmitting}
+                  type="submit"
+                  colorScheme="primary"
+                  size="sm"
+                  data-testid="save-btn"
+                >
+                  Save
+                </Button>
+              </Box>
+            </Form>
+          )}
+        </Formik>
+      </Stack>
+      {configurationStep === "apiKey" ||
+      configurationStep === "testConnection" ? (
```

Comment:
It looks like nothing ever sets `configurationStep` to `"testConnection"`. Is that another ticket? A `TODO` comment would help.